cpp/perspective/src/cpp/arrow_csv.cpp

```diff
@@ -12,9 +12,13 @@
 #include <arrow/util/value_parsing.h>
 #include <arrow/io/memory.h>
 
-// This causes build warnings
-// https://github.com/emscripten-core/emscripten/issues/8574
-#include <perspective/vendor/arrow_single_threaded_reader.h>
+#ifdef PSP_ENABLE_WASM
+    // This causes build warnings
+    // https://github.com/emscripten-core/emscripten/issues/8574
+    #include <perspective/vendor/arrow_single_threaded_reader.h>
+#else
+    #include <arrow/csv/reader.h>
```

Comment:
@texodus I can enable the `read_options.use_threads = false;` to true a few lines down - I am assuming it should be safe given that the API is exposed by Arrow and we don't do any manual concurrency for this on our side.

Reply:
Arrow seems to use [this implementation](https://github.com/apache/arrow/blob/b73bcf0a0ddacd5adc80389b10b8c1b3820ee97a/cpp/src/arrow/util/thread_pool.cc#L395) for populating the default worker pool, which will consume all hardware concurrency without proper env flags.  We should probably figure out how to override these flags in the loaded env, and to respect what was asked for in `set_num_threads()`.

Interesting though, we could easily reuse this static threadpool in place of TBB, looks like [`ParallelFor()`](https://github.com/apache/arrow/blob/b73bcf0a0ddacd5adc80389b10b8c1b3820ee97a/cpp/src/arrow/util/parallel.h#L35) is nearly a dropin replacement for `PSP_PARALLEL_FOR` ..